.github/pull_request_template.md

```diff
@@ -0,0 +1,28 @@
+# Pull Request
+
+## Description
+[Provide a brief description of the changes implemented in this pull request.]
+
+## Related Issues
+[Link any related issues or tasks from your project management system.]
+
+## Additional Notes
+[Add any additional context or notes that reviewers should know about.]
+
+## Checklist
+Please review and complete the following checklist before submitting your pull request:
+
+- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
```

Comment:
I like this idea and am wondering how the developer would ensure that the PR is a manageable size. And what the developer should do if the changes in a PR get too big. (curious because I normally don't think about the size of a PR as long as I'm fulfilling the related issue ticket)

Reply:
It could stand to have a metric attached to it for clarity, but I struggle to find a good one.

LoC changed could work, as I'm sure many of us would be unhappy to see a 10k LoC changed pull request to review.  However, there may be times when this makes sense.  Take a large project, with a simple refactor, many of those can easily hit the thousands of lines changed, but the amount of mental energy necessary to review is low.  So there are exceptions.

Hours, might be closer to what we want, but each of us likely review at different speeds.

I've intentionally left this open-ended to encourage the author(s) to approach it as a genuine question of good faith. Have you made an effort to break down this problem into manageable components, ensuring that the reviewers won't need days or weeks to comprehend the changes being made?

With all that said, if you have a suggestion to re-word, I'm all ears.

Reply:
I'm fine with the wording. Especially if its meant to be open ended. I like it as a check box to think about when you're doing a PR. Thanks!